File: backend/src/modules/employees/employee-status-log.entity.ts
import {
  Entity, PrimaryGeneratedColumn, Column,
  ManyToOne, JoinColumn, CreateDateColumn
} from 'typeorm';
import { Employee } from './employee.entity';
import { EmployeeStatus } from '../../common/enums';

@Entity('employee_status_logs')
export class EmployeeStatusLog {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => Employee, employee => employee.statusLogs, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'employee_id' })
  employee: Employee;

  @Column({
    type: 'varchar',
    length: 20,
  })
  status: EmployeeStatus;

  @Column({ name: 'start_date', type: 'date' })
  startDate: Date;

  @Column({ name: 'end_date', type: 'date', nullable: true })
  endDate: Date;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;
}

File: lib/features/profile/domain/entities/employee_status_log_entity.dart
import 'package:equatable/equatable.dart';

class EmployeeStatusLogEntity extends Equatable {
  final String id;
  final String status;
  final DateTime startDate;
  final DateTime? endDate;
  final DateTime createdAt;

  const EmployeeStatusLogEntity({
    required this.id,
    required this.status,
    required this.startDate,
    this.endDate,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, status, startDate, endDate, createdAt];
}
